python/tk_multi_breakdown2/utils.py

```diff
@@ -47,7 +57,42 @@ def get_ui_published_file_fields(app):
     fields += utils.resolve_sg_fields(file_history_config.get("top_left"))
     fields += utils.resolve_sg_fields(file_history_config.get("top_right"))
     fields += utils.resolve_sg_fields(file_history_config.get("body"))
-    if file_history_config["thumbnail"] and "image" not in fields:
-        fields.append("image")
+    if file_history_config["thumbnail"]:
+        if "image" not in fields:
+            fields.append("image")
+        # We need the linked Version's image if setting
+        # use_version_thumbnail_as_fallback is enabled.
+        if "version.Version.image" not in fields:
+            fields.append("version.Version.image")
 
     return list(set(fields))
+
+
+def get_item_image_field(item):
+    """
+    Get the field to use for the thumbnail for the given item.
+
+    It returns "image" if the item has an image field and there's an
+    actual image to use. If there's no image field or no image to use,
+    it tries to use the version.Version.image field in the same way,
```

Comment:
```suggestion
    it tries to use the `version.Version.image` field in the same way,
```

Reply:
I think this docstring is too close to the implementation and could be simply: 

Check if a thumbnail is available for the given item, and return the field name to download it from.
If not available and falling back on the Version is allowed, check if one is available for it and, if so, return the dotted field to use to retrieve it.

Please note that I think that `fallback_on_version_thumbnail` should be passed as a param.